src/components/clusterprovider/azure/azure.ts

```diff
@@ -0,0 +1,388 @@
+'use strict';
+
+import { Shell } from '../../../shell';
+import { FS } from '../../../fs';
+import { Errorable, ActionResult, fromShellJson, fromShellExitCode } from '../../../wizard';
+import * as compareVersions from 'compare-versions';
+import { sleep } from '../../../sleep';
+
+export interface Context {
+    readonly fs: FS;
+    readonly shell: Shell;
+}
+
+export interface ServiceLocation {
+    readonly displayName: string;
+    readonly isPreview: boolean;
+}
+
+export interface Locations {
+    readonly locations: any;
+}
+
+export interface LocationRenderInfo {
+    readonly location: string;
+    readonly displayText: string;
+}
+
+export interface ClusterInfo {
+    readonly name: string;
+    readonly resourceGroup: string;
+}
+
+export interface ConfigureResult {
+    readonly gotCli: boolean;
+    readonly cliInstallFile: string;
+    readonly cliOnDefaultPath: boolean;
+    readonly cliError: string;
+    readonly gotCredentials: boolean;
+    readonly credentialsError: string;
+}
+
+export interface WaitResult {
+    readonly stillWaiting?: boolean;
+}
+
+const MIN_AZ_CLI_VERSION = '2.0.23';
+
+export async function getSubscriptionList(context: Context, forCommand: string) : Promise<ActionResult<string[]>> {
+    // check for prerequisites
+    const prerequisiteErrors = await verifyPrerequisitesAsync(context, forCommand);
+    if (prerequisiteErrors.length > 0) {
+        return {
+            actionDescription: 'checking prerequisites',
+            result: { succeeded: false, result: [], error: prerequisiteErrors }
+        };
+    }
+
+    // list subs
+    const subscriptions = await listSubscriptionsAsync(context);
+    return {
+        actionDescription: 'listing subscriptions',
+        result: subscriptions
+    };
+}
+
+async function verifyPrerequisitesAsync(context: Context, forCommand: string) : Promise<string[]> {
+    const errors = new Array<string>();
+    
+    const azVersion = await azureCliVersion(context);
+    if (azVersion === null) {
+        errors.push('Azure CLI 2.0 not found - install Azure CLI 2.0 and log in');
+    } else if (compareVersions(azVersion, MIN_AZ_CLI_VERSION) < 0) {
+        errors.push(`Azure CLI required version is ${MIN_AZ_CLI_VERSION} (you have ${azVersion}) - you need to upgrade Azure CLI 2.0`);
+    }
+
+    if (forCommand == 'acs') {
+        prereqCheckSSHKeys(context, errors);
+    }
+
+    return errors;
+}
+
+async function azureCliVersion(context: Context) : Promise<string> {
```

Comment:
i feel like we should use https://github.com/npm/node-semver here

Reply:
We currently use https://www.npmjs.com/package/compare-versions.  I'm not sure what `semver` offers to justify switching.

Reply:
argh derp;  i forgot that the output for `az --version` looks like:

```
azure-cli (2.0.27)

acr (2.0.21)
acs (2.0.26)
advisor (0.1.2)
appservice (0.1.26)
backup (1.0.6)
batch (3.1.10)
batchai (0.1.5)
billing (0.1.7)
cdn (0.0.13)
cloud (2.0.12)
cognitiveservices (0.1.10)
command-modules-nspkg (2.0.1)
configure (2.0.14)
consumption (0.2.1)
container (0.1.18)
core (2.0.27)
cosmosdb (0.1.19)
dla (0.0.18)
dls (0.0.19)
eventgrid (0.1.10)
extension (0.0.9)
feedback (2.1.0)
find (0.2.8)
interactive (0.3.16)
iot (0.1.17)
keyvault (2.0.18)
lab (0.0.17)
monitor (0.1.2)
network (2.0.23)
nspkg (3.0.1)
profile (2.0.19)
rdbms (0.0.12)
redis (0.2.11)
reservations (0.1.1)
resource (2.0.23)
role (2.0.19)
servicefabric (0.0.10)
sql (2.0.21)
storage (2.0.25)
vm (2.0.26)

Python location '/usr/local/opt/python3/bin/python3.6'
Extensions directory '/Users/bhargav/.azure/cliextensions'

Python (Darwin) 3.6.4 (default, Jan 23 2018, 14:22:14)
[GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)]

Legal docs and information: aka.ms/AzureCliLegal
````
… nvm my suggestion then - i was wondering why you were using the regex parsing, and now i get it.